### PR TITLE
Update authority command checks for weight zero

### LINKED ISSUE #8929: Update verify logic in authority module update authority command
### Expected behavior

Verify hook in update authority command should have proper schema and checks

### Actual behavior

- Validator length check can be done in schema. 
- totalWeight === 0 check missing
- Validator weight cannot be 0

### Steps to reproduce

Check verify hook logic

### Which version(s) does this affect? (Environment, OS, etc...)
v6.0.0

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #8929 

### How was it solved?

- Added zero checks
- Updated verify hook to return object instead of throw error

### How was it tested?

Updated tests

### CODE REVIEW

File: framework/src/modules/poa/commands/update_authority.ts [hunk 1]
```diff
@@ -54,50 +53,87 @@ export class UpdateAuthorityCommand extends BaseCommand {
 		context: CommandVerifyContext<UpdateAuthorityParams>,
 	): Promise<VerificationResult> {
 		const { newValidators, threshold, validatorsUpdateNonce } = context.params;
-
-		if (newValidators.length < 1 || newValidators.length > MAX_NUM_VALIDATORS) {
-			throw new Error(
-				`newValidators length must be between 1 and ${MAX_NUM_VALIDATORS} (inclusive).`,
-			);
+		try {
+			validator.validate(updateAuthoritySchema, context.params);
```
Comment (sergeyshemyakov) [inline framework/src/modules/poa/commands/update_authority.ts:57]: I think the schema validity is checked separately in the state machine itself, see [here](https://github.com/LiskHQ/lisk-sdk/pull/8558/files#diff-cb2314f09fb36abaa9b56f7e516e6753a48441c7e8ac30478fd6ad2a9f998360).

**Review by sergeyshemyakov — APPROVED** [review]
I have just one comment, otherwise the PR looks good

File: framework/src/modules/poa/commands/update_authority.ts [hunk 2]
```diff
@@ -54,50 +53,79 @@ export class UpdateAuthorityCommand extends BaseCommand {
...
+					),
+				};
 			}
+
+			if (newValidator.weight === BigInt(0)) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error(`Validator weight cannot be zero.`),
+				};
+			}
```
Comment (ishantiw) [inline framework/src/modules/poa/commands/update_authority.ts:90]: I was wondering if we can get it checked in schema by adding minimum as 1

**Review by ishantiw — APPROVED** [review]
Overall looks good, small comment though

**Pull request merged**